pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -224,6 +263,36 @@ func createDynamicMetadataMatcher(k, v string) *metadata.MetadataMatcher {
 			},
 		}
 	}
+	return stringMatcher
+}
+
+// createDynamicMetadataMatcher creates a MetadataMatcher for the given key, value pair.
+func createDynamicMetadataMatcher(k, v string) *metadata.MetadataMatcher {
+	keys, err := generateMetaKeys(k)
+	if err != nil {
+		rbacLog.Errorf("failed to generate meta matcher key %s, err: %v", k, err)
+		return nil
+	}
+	forceRegexPattern := false
+	if k == attrSrcNamespace {
+		// Change the value to a regular expression to match the namespace part.
+		v = fmt.Sprintf(`*/ns/%s/*`, v)
+		forceRegexPattern = true
+	}
+
+	// Handle the claims under authDerivedClaims
+	if len(keys) == 2 && keys[0] == authDerivedClaims {
+		switch keys[1] {
+		case groupsClaim:
```

Comment:
We do not need to special handle "groups" claim. Any claim should generate ListMatcher.

Reply:
Done.